packages/block-editor/src/components/block-toolbar/style.scss

```diff
@@ -203,6 +203,24 @@
 
 	.block-editor-block-mover .block-editor-block-mover__move-button-container {
 		width: auto;
+
+		@include break-small() {
+			position: relative;
+
+			&::before {
+				content: "";
+				display: block;
+				height: $border-width;
+				width: 100%;
+				background: $gray-900;
+				position: absolute;
+				top: 50%;
+				left: 50%;
+				// With Top toolbar on, this separator has a smaller width. Translating both
+				// X and Y axis allows to make the separator alwasy centered regardless of its width.
+				transform: translate(-50%);
```

Comment:
```suggestion
				transform: translate(-50%, 0);
				margin-top: -$border-width * 0.5;
```

I think `translate(-50%);` means `translate(-50%, 0);`. Strictly speaking, it is not vertically centered.

I tried changing it to `translate(-50%, -50%)` to align it vertically, but the border seems to bleed.

![image](https://github.com/WordPress/gutenberg/assets/54422211/a07b9dc2-eba8-4355-91d3-fe82141825c3)

Instead, I tried using a negative margin to shift the border up by half the height value. This approach draws a solid border in my environment. Could you test if this approach works in your environment?

Reply:
> I think translate(-50%); means translate(-50%, 0);. Strictly speaking, it is not vertically centered.

Right.

> the border seems to bleed.

Are you on Windows or some GNU-Linux distro? What browser are you using? What display type and resolution?

> I tried using a negative margin to shift the border up by half the height value. This approach draws a solid border in my environment.

The root problem is that the toolbar height is 48px. The mover buttons height is 24px. A vertically centered 1 pixel horizontal line would need to be placed at exactly 23.5 pixels. However, a physical pixel can't be split in two. Some operating systems / browsers / displays (retina resolution, sub-pixel rendering) may render that more nicely than others. The 'bleeding' depends on the ability to position an element to 'half a pixel'. In environments where a CSS pixel is actually more than one physical pixel that may work better. In your screenshot, the horizontal line is actually 2 pixels, apparently that's because your environment isn't able to render sub-pixels.

The top margin approach actually computes to `margin-top: -0.5px;`. In your environment it may preserve the height of the line to 1 pixel but I guess the line isn't vertically centered.

Environments that support true subpixel rendering won't trigger this issue. The only reliable way to avoid a greyish blurred line or its rendered height to be altered would be to make sure it is moved to a 'full pixel' e.g. by making the toolbar height an odd number.

Maybe `translateZ(0)` may help prevent the blurriness because it promotes the element to its own layer but there's no guarantee the line will be vertically centered. Also `margin-top: -0.5px;` doesn't guarantee the line will be vertically centered but I'd think that's batter than a blurred line.

For the records, this is how the translateY approach looks in my environment (zoomed in): no blurring or doubled height:

![Screenshot 2024-01-11 at 10 45 07](https://github.com/WordPress/gutenberg/assets/1682452/9379a9e0-3d96-418c-b754-b3312a3ad999)